syntax = "proto3";

package Protos;

message SensorData {
    string pacifier_id = 1;
    string sensor_type = 2;
    string sensor_group = 3;
    map<string, bytes> data_map = 4;
}

message IMUData {
    message gyro {
        float gyro_x = 1;
        float gyro_y = 2;
        float gyro_z = 3;
    }
    repeated gyro gyros = 1;

    message mag {
        float mag_x = 1;
        float mag_y = 2;
        float mag_z = 3;
    }
    repeated mag mags = 2;

    message acc {
        float acc_x = 1;
        float acc_y = 2;
        float acc_z = 3;
    }
    repeated acc accs = 3;
}

message PPGData {
    message led {
    int32 led_1 = 1;
    int32 led_2 = 2;
    int32 led_3 = 3;
    }
    repeated led leds = 1;

    message temperature {
        float temperature_1 = 1;
    }
    repeated temperature temperatures = 2;

}


